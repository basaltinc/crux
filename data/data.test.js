const {
  getColors,
  getTransitions,
  getBreakpoints,
  getSpacings,
  getTypography,
} = require('./index');

describe('Data Get Functions', () => {
  test('Colors', async () => {
    expect(await getColors()).toMatchSnapshot();
    expect(await getColors('rgb')).toMatchSnapshot();
    expect(await getColors('hex')).toMatchSnapshot();
  });

  test('Transitions', async () => {
    expect(await getTransitions()).toMatchSnapshot();
  });

  test('Breakpoints', async () => {
    expect(await getBreakpoints()).toMatchSnapshot();
  });

  test('getSpacings', async () => {
    expect(await getSpacings()).toMatchSnapshot();
  });

  test('getTypography', async () => {
    expect(await getTypography()).toMatchSnapshot();
  });
});
