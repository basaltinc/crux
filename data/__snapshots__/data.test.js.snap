// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Get Functions Breakpoints 1`] = `
Array [
  Object {
    "comment": "",
    "name": "$bp--xsmall",
    "value": "380px",
  },
  Object {
    "comment": "",
    "name": "$bp--small",
    "value": "450px",
  },
  Object {
    "comment": "",
    "name": "$bp--medium",
    "value": "700px",
  },
  Object {
    "comment": "",
    "name": "$bp--large",
    "value": "900px",
  },
  Object {
    "comment": "",
    "name": "$bp--xlarge",
    "value": "1100px",
  },
  Object {
    "comment": "",
    "name": "$bp--xxlarge",
    "value": "1300px",
  },
  Object {
    "comment": "",
    "name": "$bp--xxxlarge",
    "value": "1600px",
  },
]
`;

exports[`Data Get Functions Colors 1`] = `
Array [
  Object {
    "comment": "",
    "name": "$c-blue--light",
    "value": "hsl(165, 26%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-blue",
    "value": "hsl(200, 55%, 19%)",
  },
  Object {
    "comment": "",
    "name": "$c-green",
    "value": "hsl(159, 20%, 17%)",
  },
  Object {
    "comment": "",
    "name": "$c-iron",
    "value": "hsl(240, 3%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-yellow",
    "value": "hsl(52, 74%, 54%)",
  },
  Object {
    "comment": "",
    "name": "$c-white",
    "value": "hsl(0, 0%, 100%)",
  },
  Object {
    "comment": "",
    "name": "$c-gray--dark",
    "value": "hsl(0, 0%, 40%)",
  },
  Object {
    "comment": "",
    "name": "$c-black",
    "value": "hsl(0, 0%, 0%)",
  },
]
`;

exports[`Data Get Functions Colors 2`] = `
Array [
  Object {
    "comment": "",
    "name": "$c-blue--light",
    "value": "hsl(165, 26%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-blue",
    "value": "hsl(200, 55%, 19%)",
  },
  Object {
    "comment": "",
    "name": "$c-green",
    "value": "hsl(159, 20%, 17%)",
  },
  Object {
    "comment": "",
    "name": "$c-iron",
    "value": "hsl(240, 3%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-yellow",
    "value": "hsl(52, 74%, 54%)",
  },
  Object {
    "comment": "",
    "name": "$c-white",
    "value": "hsl(0, 0%, 100%)",
  },
  Object {
    "comment": "",
    "name": "$c-gray--dark",
    "value": "hsl(0, 0%, 40%)",
  },
  Object {
    "comment": "",
    "name": "$c-black",
    "value": "hsl(0, 0%, 0%)",
  },
]
`;

exports[`Data Get Functions Colors 3`] = `
Array [
  Object {
    "comment": "",
    "name": "$c-blue--light",
    "value": "hsl(165, 26%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-blue",
    "value": "hsl(200, 55%, 19%)",
  },
  Object {
    "comment": "",
    "name": "$c-green",
    "value": "hsl(159, 20%, 17%)",
  },
  Object {
    "comment": "",
    "name": "$c-iron",
    "value": "hsl(240, 3%, 85%)",
  },
  Object {
    "comment": "",
    "name": "$c-yellow",
    "value": "hsl(52, 74%, 54%)",
  },
  Object {
    "comment": "",
    "name": "$c-white",
    "value": "hsl(0, 0%, 100%)",
  },
  Object {
    "comment": "",
    "name": "$c-gray--dark",
    "value": "hsl(0, 0%, 40%)",
  },
  Object {
    "comment": "",
    "name": "$c-black",
    "value": "hsl(0, 0%, 0%)",
  },
]
`;

exports[`Data Get Functions Transitions 1`] = `
Array [
  Object {
    "comment": "",
    "name": "$trans-opacity__time",
    "value": "0.3s",
  },
  Object {
    "comment": "",
    "name": "$trans-opacity__function",
    "value": "ease-in",
  },
  Object {
    "comment": "convenience variable for using both",
    "name": "$trans-opacity",
    "value": "$trans-opacity__time $trans-opacity__function",
  },
  Object {
    "comment": "",
    "name": "$trans-move__time",
    "value": "0.3s",
  },
  Object {
    "comment": "",
    "name": "$trans-move__function",
    "value": "ease-in",
  },
  Object {
    "comment": "convenience variable for using both",
    "name": "$trans-move",
    "value": "$trans-opacity__time $trans-move__function",
  },
]
`;

exports[`Data Get Functions getSpacings 1`] = `
Array [
  Object {
    "comment": "",
    "name": "$spacing--xs",
    "value": ".5rem",
  },
  Object {
    "comment": "",
    "name": "$spacing--s",
    "value": ".75rem",
  },
  Object {
    "comment": "",
    "name": "$spacing--m",
    "value": "1rem",
  },
  Object {
    "comment": "",
    "name": "$spacing--l",
    "value": "1.5rem",
  },
  Object {
    "comment": "",
    "name": "$spacing--xl",
    "value": "2rem",
  },
  Object {
    "comment": "",
    "name": "$spacing--xxl",
    "value": "3.3rem",
  },
  Object {
    "comment": "",
    "name": "$spacing--xxxl",
    "value": "5rem",
  },
]
`;

exports[`Data Get Functions getTypography 1`] = `
Object {
  "fontFamilies": Array [
    Object {
      "comment": "Avenir Light",
      "name": "$ff--avenir--light",
      "value": "'AvenirLight', sans-serif",
    },
    Object {
      "comment": "Avenir Medium",
      "name": "$ff--avenir--medium",
      "value": "'AvenirMedium', sans-serif",
    },
    Object {
      "comment": "Din Condensed",
      "name": "$ff--din-condensed",
      "value": "'DinCondensed', sans-serif",
    },
  ],
  "fontSizes": Array [
    Object {
      "comment": "",
      "name": "$fs--xxl",
      "value": "2.25rem",
    },
    Object {
      "comment": "",
      "name": "$fs--xl",
      "value": "1.75rem",
    },
    Object {
      "comment": "",
      "name": "$fs--l",
      "value": "1.25rem",
    },
    Object {
      "comment": "",
      "name": "$fs--m",
      "value": "1rem",
    },
    Object {
      "comment": "",
      "name": "$fs--s",
      "value": ".875rem",
    },
    Object {
      "comment": "",
      "name": "$fs--xs",
      "value": ".75rem",
    },
    Object {
      "comment": "",
      "name": "$fs--xxs",
      "value": ".625rem",
    },
  ],
}
`;
